### ISSUE TITLE
Update SDK to upstream

### ISSUE THREAD

**Issue by xan1242** [ThirteenAG/pspsdk#3]
This SDK is a version from around 2018 compiled with mingw for win32 `psp-gcc (GCC) 8.2.0`

The latest version is: `psp-gcc (GCC) 11.2.0`

Upstream: https://github.com/pspdev/pspdev

Extra libraries for hooking (such as `systemctrl.h`) from need to be taken from CFW projects such as this one: https://github.com/uofw/uofw

I tried compiling with MSYS2 but both gdb and gcc failed to compile because of some changes in the GNU unistd library, causing things to break.

There's 2 ways to go on about this:

1. Keep on trying with mingw32/MSYS/cygwin, maybe with an equivalent version of GCC
2. Scrap compiling from Windows altogether and make VS spin up a Docker container with the SDK for anything it needs to compile and perform the compilation in the VM

This is beneficial to us because PRXs act a bit weird on both PPSSPP and real hardware (such as instability, freezing or not launching at all). This _probably_ fixes those weird problems.

**Comment by ThirteenAG** [comment]
I'd really like to keep compiling on windows as an option tho.
Also, maybe it's possible to use WSL for upstream sdk?

**Comment by xan1242** [comment]
OK I did it in cygwin.

And the results are as follows:

- `make` properly redirects to `/cygdrive/driveletter/path`
- PRX files are now substantially bigger - went up from about 32KB to about 121KB - there are a lot more functions in the text segment (went from 90 to 262 in my example)
- PRX files work only in the emulator and not on real PSP

And a few notes about cygwin:

- certain (key) packages in cygwin are unmaintained; 
- libiconv was missing the pkg-config, had to make it manually
- libgpgme is version 1.9 where it expects at least 1.11 (`GPGME_KEYLIST_MODE_LOCATE` is missing and has to be replaced with `GPGME_KEYLIST_MODE_LOCAL | GPGME_KEYLIST_MODE_EXTERN` for libalpm and pacman to compile)
- meson.build for the buildpath has an extra slash in the line with `mkdir -p "$DESTDIR` which is incompatible with cygwin for whatever reason
- `python3-virtualenv` is not being installed by default, you need to install this
- for `pacman.conf` you need to enable wget as the XferCommand
- gcc stage 2 can randomly segfault during compilation

Overall was a nightmare to figure out with very middling results. I'll check it out in Linux as well, but I have a feeling like this might not be a good idea after all...

Here you go a link to the binaries if you want to play with it eventually: https://1fichier.com/?u32fgqrgsw85uqxcdxhv

If Linux builds of the toolchain don't produce any better results, I will give up on this then.

**Comment by ThirteenAG** [comment]
Yeah, tbh I can live with short and spaceless paths, because I don't know much about linux environment.

**Comment by xan1242** [comment]
Nah, worst case scenario I'll rip out the new `make` command and it'll fix the spacing issue.

I was hoping upstream would benefit us in other ways...

As for Linux, I'll try to set up a Docker container if we have to use it

**Comment by ThirteenAG** [comment]
> As for Linux, I'll try to set up a Docker container if we have to use it

How does that work exactly? Wouldn't it require a lot of dependencies?

**Comment by xan1242** [comment]
It shouldn't require more than WSL I think.

https://docs.docker.com/desktop/install/windows-install/

TLDR: it can use both WSL or Hyper-V as a backend to run a VM to perform the compilation task. The only dependency would be enabled Hyper-V in Windows (either directly or via WSL).

And since it's directly supported by the pspdev upstream (and since it's portable), it's the ideal solution in my opinion.

https://hub.docker.com/r/pspdev/pspdev/tags

**Comment by xan1242** [comment]
I'd only use this if it gives any benefits, which I'll check in Linux directly myself. If it is that much better then I'll look into it, but atm I think we'll end up just fixing the make issue and that's it...

**Comment by xan1242** [comment]
Oh I noticed that someone played with the `build_prx.mak` file and removed the debug libs in this repo... That explains the change in size and behavior.

I'll investigate further later, I'm too tired at the moment.

**Comment by xan1242** [comment]
OK so I've figured pretty much everything out.

The new SDK I sent is absolutely fine! The issues with CFW on a PSP were completely unrelated.

The module size situation can be remedied partially by some new options in the SDK to exclude the newlib library.

I'll be updating this SDK shortly. No need for Powershell or any of this stuff, it'll all work out nicely.

EDIT: actually the space issue plagues the PATH variable updating lol, Powershell will be needed after all

**Comment by ThirteenAG** [comment]
Impressive, very nice.
